src/NUnitFramework/framework/Schemas/TestFilterDefinitions.xsd

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xs:schema elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:annotation>
+    <xs:documentation>This is document is descriptive, rather than a guarantee of future decisions.</xs:documentation>
+  </xs:annotation>
+
+  <!-- NUnit.Framework.Internal.TestFilter.EmptyFilter.AddToXml -->
+  <!-- NUnit.Framework.Internal.TestFilter.FromXML -->
+  <xs:complexType name="TestFilterType">
+    <xs:group ref="TestFilterElementGroup" minOccurs="0" maxOccurs="unbounded" />
+  </xs:complexType>
+
+  <xs:group name="TestFilterElementGroup">
+    <xs:choice>
+
+      <!-- NUnit.Framework.Internal.Filters.NotFilter.AddToXml -->
+      <xs:element name="not">
+        <xs:complexType>
+          <xs:group ref="TestFilterElementGroup" />
+        </xs:complexType>
+      </xs:element>
+
+      <!-- NUnit.Framework.Internal.Filters.AndFilter.ElementName -->
+      <xs:element name="and" type="CompositeFilterType" />
+
+      <!-- NUnit.Framework.Internal.Filters.OrFilter.ElementName -->
+      <xs:element name="or" type="CompositeFilterType" />
+
+      <!-- NUnit.Framework.Internal.Filters.CategoryFilter.ElementName -->
+      <xs:element name="cat" type="ValueMatchFilterType" />
+
+      <!-- NUnit.Framework.Internal.Filters.ClassNameFilter.ElementName -->
+      <xs:element name="class" type="ValueMatchFilterType" />
+
+      <!-- NUnit.Framework.Internal.Filters.FullNameFilter.ElementName -->
+      <xs:element name="test" type="ValueMatchFilterType" />
+
+      <!-- NUnit.Framework.Internal.Filters.IdFilter.ElementName -->
+      <xs:element name="id" type="ValueMatchFilterType" />
```

Comment:
I see:

https://github.com/nunit/nunit/blob/f609611e5fc3980078376615fc6b016cc6cac1c2/src/NUnitFramework/framework/Internal/Filters/IdFilter.cs#L46-L48

The current behavior of the framework is to ignore the `IsRegex` property inherited from `ValueMatchFilter`. It doesn't error if it is set. I'm not opposed to making it invalid XML so long as we make sure the framework is changed to prevent `IdFilter.IsRegex` from being written out into the XML.

Reply:
I think I got it backwards (that the framework cannot parse e.g. `Id != ...` or `Id =~ ...` and that it would not generate a `<id re="1">...` element as the code is now).

I think we should keep the schema as it is. Then we can always improve on it later on, if we agree to enforce a more strict schema.

Reply:
Okay, sounds good.